api-analysis.md

```diff
@@ -0,0 +1,137 @@
+GET /api/v1/participants/npq
+GET /api/v1/participants/npq/{id}
+PUT /api/v1/participants/npq/{id}/change-schedule
+PUT /api/v1/participants/npq/{id}/defer
+PUT /api/v1/participants/npq/{id}/defer
+PUT /api/v1/participants/npq/{id}/resume
+PUT /api/v1/participants/npq/{id}/withdraw
+GET /api/v1/participants/npq/outcomes
+GET /api/v1/participants/npq/{participant_id}/outcomes
+POST /api/v1/participants/npq/{participant_id}/outcomes
+GET /api/v1/participant-declarations
+```
+
+### V2
+
+```
+GET  /api/v2/npq-applications
+GET  /api/v2/npq-applications.csv
+GET  /api/v2/npq-applications/{application_id}
+POST /api/v2/npq-applications/{application_id}/accept
+POST /api/v2/npq-applications/{application_id}/reject
+GET  /api/v2/npq-enrolments.csv
+GET  /api/v2/participants/npq
+GET  /api/v2/participants/npq/{participant_id}
+PUT  /api/v2/participants/npq/{participant_id}/change-schedule
+PUT  /api/v2/participants/npq/{participant_id}/defer
+PUT  /api/v2/participants/npq/{participant_id}/resume
+PUT  /api/v2/participants/npq/{participant_id}/withdraw
+GET  /api/v2/participants/npq/outcomes
+GET  /api/v2/participants/npq/{participant_id}/outcomes
+POST /api/v2/participants/npq/{participant_id}/outcomes
+GET /api/v3/participant-declarations
+```
+
+### V3
+
+```
+GET  /api/v3/npq-applications
+GET  /api/v3/npq-applications/{application_id}
+POST /api/v3/npq-applications/{application_id}/accept
+POST /api/v3/npq-applications/{application_id}/reject
+GET  /api/v3/participants/npq/outcomes
+GET  /api/v3/participants/npq/{participant_id}/outcomes
+POST /api/v3/participants/npq/{participant_id}/outcomes
+GET  /api/v3/participants/npq
+GET  /api/v3/participants/npq/{participant_id}
+PUT  /api/v3/participants/npq/{participant_id}/change-schedule
+PUT  /api/v3/participants/npq/{participant_id}/defer
+PUT  /api/v3/participants/npq/{participant_id}/resume
+PUT  /api/v3/participants/npq/{participant_id}/withdraw
+GET /api/v3/participant-declarations
+```
+
+### Identifiers: technical analysis
+
+The following analysis applies to the endpoints of the different API versions: v1, v2 and v3.
+
+#### API endpoints that receive an `application_id`
+
+```
+# samples
+GET  /api/v3/npq-applications/{application_id}
+POST /api/v3/npq-applications/{application_id}/accept
+POST /api/v3/npq-applications/{application_id}/reject
+```
+
+The parameter `application_id`:
+
+1. Is generated by ECF.
+2. Is [stored in the ￼`ecf_idf`￼](https://github.com/DFE-Digital/npq-registration/blob/452b3831254eb32c3e72c6118ee2980a1df72b6f/db/schema.rb#L32) in NPQ
+3. Is returned to Lead Providers via [API endpoint response](%5Bresponse%20of%20this%20endpoint%5D%28https%3A//manage-training-for-early-career-teachers.education.gov.uk/api-reference/reference-v3.html#api-v3-npq-applications-id-accept-post-parameters%29).
+
+##### Actions
+
+1. No need to migrate ECF `application_id` as this information is already present in the attribute: `ecf_id` in the `applications` table in NPQ.
```

Comment:
discussed in person: there is a bit of data misalignment between ECF/NPQ after that is resolved in the data migration we can trust this field to be true

Reply:
same below